### ISSUE TITLE
nc file locked by da.compute() operation

### ISSUE THREAD

**Issue by delandmeterp** [dask/dask#4966]
Hi,
I've recently started to use dask.array to compute and load only chunks of a huge dataset for the development of [Parcels](https://github.com/OceanParcels/parcels). I've run into an issue of locked netcdf files while running multiple tests using py.test, and I think my bug should be related to the reduced issue that I reproduce hereinafter:

```
import numpy as np
import xarray as xr
import dask.array as da

def create_ds():
    temp = 15 + 8 * np.random.randn(2, 2)
    precip = 10 * np.random.rand(2, 2)
    lon = [[-99.83, -99.32], [-99.79, -99.23]]
    lat = [[42.25, 42.21], [42.63, 42.59]]
    ds = xr.Dataset({'temperature': (['x', 'y'],  temp),
                     'precipitation': (['x', 'y'], precip)},
                      coords={'lon': (['x', 'y'], lon),
                              'lat': (['x', 'y'], lat)})
    ds.to_netcdf('test.nc')

def dask_op():
    ds = xr.open_dataset('test.nc')
    temp = da.from_array(ds.temperature, chunks='auto')
    temp.compute()
    temp.compute()

create_ds()
dask_op()
dset = xr.Dataset()
dset.to_netcdf('test.nc')
```

Is it normal that `test.nc` is still locked after returning from `dask_op` function?
I know the example does not crash anymore if I call `temp.compute()` only once or if I close the dataset after last computing. And if I have closed earlier the ds (like before compute()) it does not work anymore.
But is it normal I should close the the dataset explicitly?

So to summarise this works

```
temp.compute()
```
```
temp.compute()
temp.compute()
ds.close
``` 

And This doesn't
```
temp.compute()
temp.compute()
```
```
ds.close
temp.compute()
temp.compute()
ds.close
```

Doest that make sense to someone? Is it expected or a bug?

Thanks!
Philippe

**Comment by martindurant** [comment]
I am not sure about the locking, but but can you use open_dataset with chunks, so that the array is there from the start `ds.temperature.values`. There may be a bug here, but I would first check with xaray for the right usage pattern.

**Comment by delandmeterp** [comment]
Thanks! I've replaced

```
    ds = xr.open_dataset('test.nc')
    temp = da.from_array(ds.temperature, chunks='auto')
```
by
```
    ds = xr.open_dataset('test.nc', chunks={'x': -1, 'y': -1})
    temp = ds.temperature
```
but the issue occurs as mentioned in my initial comment. (by the way, while opening directly with chunks, I can't use anymore the chunks='auto' flag).

**Comment by martindurant** [comment]
I suggest asking xarray, since this is a common usage pattern for them, and I haven't heard of a drop in performance.

**Comment by martindurant** [comment]
@delandmeterp , did you raise any xarray issue, or are you convinced that there is something on our side?

**Comment by delandmeterp** [comment]
No, I'm not convinced of anything. I'll raise an xarray issue as you suggest.
